Feat : OW-60 회원가입 추가

Dongjin113 opened the pull request:
- [x] 회원가입시 중복가입 확인
- [x] 이메일 인증 확인
- [x] 이메일 인증이후 30분 이내 가입 확인
- [x] 회원가입시 비밀번호 암호화
- [x] UserRequest Dto -> SignUpRequest로 변경
- [x] User와 EmailAuth 연관관계 추가
- [x] SignUpSaveDto User에 저장할 때 사용할  Dto 추가


### Feat : OW-67 로그인 추가 , 비밀번호 변경 암호화

### 로그인 추가
- 로그인 시 refreshToken(2주) 쿠키에 accessToken(5분) 헤더에 발급 됌

### 비밀번호 변경
- 비밀번호 변경시 암호화된 비밀번호 비교 후 다시 암호화되어 저장

Dongjin113 committed:
Feat OW-60 : 회원가입 추가

- [ ] 회원가입시 중복가입 확인
- [ ] 이메일 인증 확인
- [ ] 이메일 인증이후 30분 이내 가입 확인

dunowljj commented on back/src/main/java/com/ogjg/back/user/service/UserService.java:
```diff
@@ -57,6 +64,28 @@ protected User findByEmail(String email) {
+
+        EmailAuth emailAuth = findEmailAuthByEmail(userRequest.getEmail());
+
+        if (!emailAuth.isStatus() || emailAuth.getAuthenticatedAt() == null){
+            throw new SignUpFailure("이메일 인증을 진행해 주세요");
+        }
+
+        long minutes = Duration.between(emailAuth.getAuthenticatedAt(), LocalDateTime.now()).toMinutes();
+        if (minutes > 30){
+            throw new SignUpFailure("이메일 인증을 다시 진행해 주세요");
+        }
```
두 에러 메시지의 의미가 구분하기 모호하단 생각이 들었어요. 에러 메시지를 좀 더 구체적으로 써주시면 좋을 것 같습니다!

Dongjin113 replied:
변경했습니다

dunowljj commented on back/src/main/java/com/ogjg/back/user/service/UserService.java:
```diff
@@ -57,6 +64,28 @@ protected User findByEmail(String email) {
      * 회원가입
      * */
```
@Transactional이 누락되어있습니다!

Dongjin113 replied:
수정했숨다

Dongjin113 committed:
Feat : OW-60 회원가입 비밀번호 암호화

- [ ] 회원가입시 비밀번호 암호화
- [ ] UserRequest Dto -> SignUpRequest로 변경
- [ ] User와 EmailAuth 연관관계 추가
- [ ] SignUpSaveDto User에 저장할 때 사용할  Dto 추가

bstaran reviewed (approved):
고생하셨습니다!! 👍

dunowljj reviewed (approved):
고생하셨습니다~

Dongjin113 committed:
Feat : OW-67 로그인 추가 , 비밀번호 변경 암호화

### 로그인 추가
- 로그인 시 refreshToken(2주) 쿠키에 accessToken(5분) 헤더에 발급 됌

### 비밀번호 변경
- 비밀번호 변경시 암호화된 비밀번호 비교 후 다시 암호화되어 저장
